### PR TITLE
Migrate AWS Fargate metrics module to package

### PR DESCRIPTION
**Pull request by zmoog**
## What does this PR do?

Adds a brand new `awsfargate` package with the migration of the [awsfargate module from Metricbeat](https://github.com/elastic/beats/tree/master/x-pack/metricbeat/module/awsfargate).

As the origin module, this package supports the collection of the `task_stats` metrics from Amazon Elastic Container Service (Amazon ECS) clusters only.

Metrics from the Amazon EKS clusters are not supported by this package yet.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## How to test this PR locally

Testing locally requires a special container from AWS. 

To learn more, follow the step described at zmoog/til#7

## Related issues

- Relates #2561

### CODE REVIEW

File: packages/awsfargate/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+  - FLEET_ENROLLMENT_TOKEN: `<enrollment-token>`
+  - FLEET_URL: `<fleet-server-url>`
+  
+Tip: use the AWS Secrets Manager to store the Fleet Server enrollment token.
+
+#### Service
+
+Select an existing ECS cluster and create a new service with launch type "FARGATE". Use the task definition we just created.
+
+As soon as the Elastic Agent is started, open the dashboard named "\[Metricbeat AWS Fargate\]: Fargate Overview" and you will see the metrics show up in few minutes.
```
Comment (ravikesarwani) [inline packages/awsfargate/_dev/build/docs/README.md:56]: Suggest renaming the dashboard to "[AWS Fargate] Fargate Overview".

File: packages/awsfargate/_dev/build/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+aws cloudformation delete-stack --stack-name elastic-agent-fargate-deployment
+```
+
+## References
+
+If you want to learn more about how Metricbeat works behind the scenes:
+
+- [AWS Fargate module](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-metricset-awsfargate-task_stats.html)
+- [How to monitor Amazon ECS with Elastic Observability](https://www.elastic.co/blog/how-to-monitor-amazon-ecs-with-elastic-observability)
+
```
Comment (ravikesarwani) [inline packages/awsfargate/_dev/build/docs/README.md:275]: Remove references to metricbeat. Bring anything important from metricbeat-metricset-awsfargate-task_stats.html into here. 
The other link (How to monitor ECS ...) is a great reference to keep.

File: packages/awsfargate/_dev/build/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
+# AWS Fargate Integration
+
+The AWS Fargate integration provides a method to retrieve metadata, network metrics, and Docker stats about your containers and the tasks they are a part of an [Amazon Elastic Container Service (Amazon ECS)](https://aws.amazon.com/ecs/?pg=ln&sec=hiw) cluster.
+
+## How it works?
+
+The Elastic Agent is executed as container inside your an ECS cluster and it collects metrics using the [Amazon ECS task metadata endpoint](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-metadata-endpoint-fargate.html).
+
+The ECS task metadata endpoint is an HTTP endpoint available to each container and enabled by default on [AWS Fargate platform version 1.4.0](https://aws.amazon.com/blogs/containers/aws-fargate-launches-platform-version-1-4/) an later. The Elastic Agent uses [Task metadata endpoint version 4](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-metadata-endpoint-v4-fargate.html).
+
```
Comment (ravikesarwani) [inline packages/awsfargate/_dev/build/docs/README.md:10]: Typo "and later".

File: packages/awsfargate/_dev/build/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
+# AWS Fargate Integration
+
+The AWS Fargate integration provides a method to retrieve metadata, network metrics, and Docker stats about your containers and the tasks they are a part of an [Amazon Elastic Container Service (Amazon ECS)](https://aws.amazon.com/ecs/?pg=ln&sec=hiw) cluster.
+
```
Comment (ravikesarwani) [inline packages/awsfargate/_dev/build/docs/README.md:4]: Maybe "The AWS Fargate integration helps to retrieve metadata..."

File: packages/awsfargate/_dev/build/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+## How it works?
+
+The Elastic Agent is executed as container inside your an ECS cluster and it collects metrics using the [Amazon ECS task metadata endpoint](https://docs.aws.amazon.com/AmazonECS/latest/userguide/task-metadata-endpoint-fargate.html).
+
```
Comment (ravikesarwani) [inline packages/awsfargate/_dev/build/docs/README.md:8]: Maybe "The Elastic Agent runs in a container (as a sidecar) defined in the same task definition inside your ECS cluster and collects metrics ...".

Comment (zmoog) [reply packages/awsfargate/_dev/build/docs/README.md:8]: Oh, thank you, I did it again (passive voice)! 🤦

File: packages/awsfargate/_dev/build/docs/README.md (hunk 2 above)
Comment (zmoog) [reply packages/awsfargate/_dev/build/docs/README.md:272]: Turned the "references" section into "further readings": this name is probably a better fit for the content. I also double-checked all the essential material from the Metricbeat docs has an analogous one in this README.

File: packages/awsfargate/manifest.yml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+format_version: 1.0.0
+name: awsfargate
+title: AWS Fargate
+version: 0.1.0
+license: basic
```
Comment (zmoog) [inline packages/awsfargate/manifest.yml:5]: @ravikesarwani is this license type correct for this package?

Comment (ravikesarwani) [reply packages/awsfargate/manifest.yml:5]: Yes, all types of ingest is basic license.

Comment (zmoog) [reply packages/awsfargate/manifest.yml:5]: Thank you!

File: packages/awsfargate/_dev/build/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,279 @@
+# AWS Fargate Integration
+
+The AWS Fargate integration helps to retrieve metadata, network metrics, and Docker stats about your containers and the tasks they are a part of an [Amazon Elastic Container Service (Amazon ECS)](https://aws.amazon.com/ecs/?pg=ln&sec=hiw) cluster.
```
[the remark this answers is not in the document]
Comment (zmoog) [reply packages/awsfargate/_dev/build/docs/README.md:3]: Thanks, I missed this one! 🙇

[the remark this answers is not in the document]
Comment (zmoog) [reply packages/awsfargate/_dev/build/docs/README.md]: Fixed.

File: packages/awsfargate/data_stream/task_stats/manifest.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+type: metrics
+title: AWS Fargate task_stats metrics
+release: experimental
```
Comment (kaiyan-sheng) [inline packages/awsfargate/data_stream/task_stats/manifest.yml:3]: Do we want this data stream to be experimental while the package is beta?

Comment (zmoog) [reply packages/awsfargate/data_stream/task_stats/manifest.yml]: Updated: @kaiyan-sheng thank you for catching it!

Comment (kaiyan-sheng) [reply packages/awsfargate/data_stream/task_stats/manifest.yml]: Thanks 😬 `release` is not required in the data stream level, but it is required in the package level.

**Review by kaiyan-sheng — APPROVED** [review]
LGTM! One nit: could you compress the dashboard screenshot just in case it's too large. Reference pr: https://github.com/elastic/integrations/pull/2767

**Review by aspacca — APPROVED** [review]

**Comment by zmoog** [comment]
```shell
# before (1.2M)
$ ls -l
-rw-r--r--  1 zmoog  staff  1291721 Apr  7 16:25 metricbeat-awsfargate-overview.png

$ pngquant --ext .png --force *.png

# after (400K)
$ ls -l
-rw-r--r--  1 zmoog  staff  407842 Apr 12 10:44 metricbeat-awsfargate-overview.png
```

**Review by mtojek — APPROVED** [review]
CODEOWNERS LGTM

**Pull request merged**